src/emu/diimage.h

```diff
@@ -213,7 +213,7 @@ class device_image_interface : public device_interface
 
 	const std::string &instance_name() const { return m_instance_name; }
 	const std::string &brief_instance_name() const { return m_brief_instance_name; }
-	const std::string &cannonical_instance_name() const { return m_canonical_instance_name; }
+	const char *cannonical_instance_name() const { return dynamic_cast<const device_t *>(this)->tag(); }
```

Comment:
For what it's worth, there's a spelling mistake - "canonical" should not have a double 'n'.